session.go

```diff
@@ -580,19 +590,34 @@ func (s *Session) incomingStream(id uint32) error {
 }
 
 // closeStream is used to close a stream once both sides have
-// issued a close.
+// issued a close. If there was an in-flight SYN and the stream
+// was not yet established, then this will give the credit back.
 func (s *Session) closeStream(id uint32) {
 	s.streamLock.Lock()
+	if _, ok := s.inflight[id]; ok {
+		select {
+		case <-s.synCh:
+		default:
+			s.logger.Printf("[ERR] un-established stream without inflight syn semaphore")
```

Comment:
@ryanuber I tweaked the wording on these a bit. Since any of these errors are either logic errors or a malicious client I figured I'd log them with `ERR` visibility since it's fairly serious.